src/ext/java/org/opentripplanner/ext/vehicleparking/hslpark/HslFacilitiesDownloader.java

```diff
@@ -0,0 +1,100 @@
+package org.opentripplanner.ext.vehicleparking.hslpark;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.function.BiFunction;
+import org.opentripplanner.routing.vehicle_parking.VehicleParking;
+import org.opentripplanner.routing.vehicle_parking.VehicleParkingGroup;
+import org.opentripplanner.transit.model.framework.FeedScopedId;
+import org.opentripplanner.util.HttpUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class HslFacilitiesDownloader {
+
+  private static final Logger log = LoggerFactory.getLogger(HslFacilitiesDownloader.class);
+  private final String jsonParsePath;
+  private final BiFunction<JsonNode, Map<FeedScopedId, VehicleParkingGroup>, VehicleParking> facilitiesParser;
+  private String url;
+
+  public HslFacilitiesDownloader(
+    String url,
+    String jsonParsePath,
+    BiFunction<JsonNode, Map<FeedScopedId, VehicleParkingGroup>, VehicleParking> facilitiesParser
+  ) {
+    this.url = url;
+    this.jsonParsePath = jsonParsePath;
+    this.facilitiesParser = facilitiesParser;
+  }
+
+  public List<VehicleParking> downloadFacilities(
+    Map<FeedScopedId, VehicleParkingGroup> hubForPark
+  ) {
+    if (url == null) {
+      log.warn("Cannot download updates, because url is null!");
+      return null;
+    }
+
+    try (InputStream data = HttpUtils.openInputStream(url, null)) {
+      if (data == null) {
+        log.warn("Failed to get data from url {}", url);
+        return null;
+      }
+      return parseJSON(data, hubForPark);
+    } catch (IllegalArgumentException e) {
+      log.warn("Error parsing facilities from {}", url, e);
+    } catch (JsonProcessingException e) {
+      log.warn("Error parsing facilities from {} (bad JSON of some sort)", url, e);
+    } catch (IOException e) {
+      log.warn("Error reading facilities from {}", url, e);
+    }
+    return null;
+  }
+
+  private static String convertStreamToString(java.io.InputStream is) {
+    try (java.util.Scanner scanner = new java.util.Scanner(is).useDelimiter("\\A")) {
+      return scanner.hasNext() ? scanner.next() : "";
+    }
+  }
+
+  private List<VehicleParking> parseJSON(
+    InputStream dataStream,
+    Map<FeedScopedId, VehicleParkingGroup> hubForPark
+  ) throws IllegalArgumentException, IOException {
+    ArrayList<VehicleParking> out = new ArrayList<>();
+
+    String facilitiesString = convertStreamToString(dataStream);
+
+    ObjectMapper mapper = new ObjectMapper();
```

Comment:
I think this can be a static member variable.

Reply:
Updated in https://github.com/opentripplanner/OpenTripPlanner/pull/4510/commits/7ab98bbffe1d3f591be5244331a8a9fcd880cb4f . Is this what you wanted?